### PR TITLE
Cleanup AppDomain

### PR DESCRIPTION
**Pull request by justinvp**
Remove dead code and various cleanup.

### CODE REVIEW

File: src/mscorlib/src/System/AppDomain.cs [hunk 1]
```diff
@@ -330,18 +330,12 @@ public String BaseDirectory
...
-            if (fn != null)
-            {
-                sb.Append(SR.Loader_Name + fn);
-                sb.Append(Environment.NewLine);
-            }
-
-            return StringBuilderCache.GetStringAndRelease(sb);
+            string fn = nGetFriendlyName();
+            return fn != null ?
+                SR.Loader_Name + fn + Environment.NewLine :
```
Comment (justinvp) [inline src/mscorlib/src/System/AppDomain.cs]: Can we remove the trailing `Environment.NewLine`? Looks like the only reason it's there is from a [previous implementation](https://github.com/Microsoft/referencesource/blob/b31308b03e8bd5bf779fb80fda71f31eb959fe0b/mscorlib/system/appdomain.cs#L2210-L2233) where additional diagnostic information could be appended after it.

File: src/mscorlib/src/System/AppDomain.cs [hunk 2]
```diff
@@ -330,18 +330,12 @@ public String BaseDirectory
             }
         }
 
-        public override String ToString()
+        public override string ToString()
```
Comment (jkotas) [inline src/mscorlib/src/System/AppDomain.cs]: This is unused - AppDomain in CoreLib is internal. The actual AppDomain type lives in CoreFX.

You can delete this method.

Comment (jkotas) [reply src/mscorlib/src/System/AppDomain.cs]: IsFullyTrusted property seems to be unused as well.

**Comment by justinvp** [comment]
@jkotas, I've removed the unused code from `AppDomain` and cleaned things up. PTAL.

**Comment by justinvp** [comment]
@dotnet-bot test Ubuntu16.04 arm Cross Debug Innerloop Build please (An error occurred when trying to initialize the tools.)

**Pull request merged**